### PR TITLE
Add storybook

### LINKED ISSUE #1: [FEATURE]: Add storybook for easy UI testing
### Detailed description

Maybe we can add [storybook](https://storybook.js.org/docs/react/get-started/introduction)

### Context

It would make it easier
- display the components to users.
- help developers test the components.

### Possible implementation

_No response_

### Additional information

Let me know everyone's thoughts on this. 😄 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by AbhinavMV**
## Related Issue
- Add storybook support

Closes: #1 
## Type of change
- Added multiple packages for storybook
- Cleaned package.json file, moved most of dependencies to devDependencies
- Add rollup config

I have changed the code of HSButton.Could you check and verify if this type of code is ok instead of using the assertType function?
Also i am unable to use the colors given in type/color-preset.d.ts. i.e - i can use color like success,danger,warning but none of the other colors(and tones) are working.
Could you give an example on how to use those colors?

What sort of change have you made:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update.
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented on my code, particularly wherever it was hard to understand.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published in downstream modules.

## Screenshots (if applicable)

 Original           | Updated
 :--------------------: |:--------------------:
 original screenshot | updated screenshot |
 
## Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/WebXDAO/.github/blob/main/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: src/components/HSButton/types.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+export interface HSButtonProps {
+  label: string;
+  type?: ButtonVariant;
+  color?: ButtonColor;
```
Comment (MatteuSan) [inline src/components/HSButton/types.ts:4]: I think we should only limit the color to "success", "warning", and "danger". Then call the prop "state" to be in line with the [button api](https://hashi-docs.netlify.app/docs/next/develop/button#variant-classes) in the docs.

Comment (MatteuSan) [inline src/components/HSButton/types.ts]: I think we should only limit the color to "success", "warning", and "danger". Then call the prop "state" to be in line with the [button specifications](https://hashi-docs.netlify.app/docs/next/design/button#states) in the docs.

File: src/components/HSButton/HSButton.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import '../../styles/index.scss';
+
+import { ButtonVariant, HSButtonProps } from './types';
+
+const HSButton: React.FC<HSButtonProps> = ({
+  label = 'Click Me!',
+  icon,
+  type = null,
+  color = null,
```
Comment (MatteuSan) [inline src/components/HSButton/HSButton.tsx]: I think we should only limit the color to "success", "warning", and "danger". Then call the prop "state" to be in line with the [button specifications](https://hashi-docs.netlify.app/docs/next/design/button#states) in the docs.

File: src/components/HSButton/HSButton.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  link,
+  onClick,
+  children
+}) => {
+  const ButtonBase: React.ReactElement = (
+    <>
+      {icon && <i className='hs-button__icon'>{icon}</i>}
+      {label || (children && <span className='hs-button__label'>{label}</span>)}
```
Comment (MatteuSan) [inline src/components/HSButton/HSButton.tsx]: Can we have the default value of the button label to be the children? I think that makes it for a better DX. (or am I just reading the code wrong lmao)

**Review by MatteuSan — APPROVED** [review]
Everything LGTM, except for the parts with the comments. Btw, I think we should take the approach I took when I was writing out the public API: Write the public API in markdown first then implement, whenever we want to make a change, go back to writing it in markdown and do it again. This way we don't have to waste time refactoring and refactoring the API.

File: src/components/HSButton/types.ts (hunk 1 above)
Comment (AbhinavMV) [reply src/components/HSButton/types.ts]: Done this change

File: src/components/HSButton/HSButton.tsx (hunk 3 above)
Comment (AbhinavMV) [reply src/components/HSButton/HSButton.tsx]: I have removed label prop and gave children type - string | ReactChild 
I hope this is what you wanted 😄

**Comment by MatteuSan** [comment]
aight lgtm. imma merge it now

**Comment by AbhinavMV** [comment]
I have added the component in readme 
But once we have the story book setup then will deploy the storybook 
and update readme again for button component

**Pull request merged**